src/Microsoft.Azure.WebJobs.Extensions.Kafka/KafkaListener.cs

```diff
@@ -0,0 +1,229 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Avro.Generic;
+using Avro.Specific;
+using Confluent.Kafka;
+using Confluent.SchemaRegistry;
+using Confluent.SchemaRegistry.Serdes;
+using Microsoft.Azure.WebJobs.Host.Executors;
+using Microsoft.Azure.WebJobs.Host.Listeners;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Kafka
+{
+
+    internal class KafkaListener<TKey, TValue> : IListener
+    {
+        private ITriggeredFunctionExecutor executor;
+        private bool singleDispatch;
+        private readonly ILogger logger;
+        private readonly string brokerList;
+        private readonly string topic;
+        private readonly string consumerGroup;
+        private readonly string eventHubConnectionString;
+        private readonly string avroSchema;
+        private readonly int maxBatchSize;
+        private Consumer<TKey, TValue> consumer;
+        IKafkaMessagePublisher<TKey, TValue> publisher;
+        private bool disposed;
+
+        public KafkaListener(
+            ITriggeredFunctionExecutor executor,
+            bool singleDispatch,
+            ILogger logger,
+            string brokerList,
+            string topic,
+            string consumerGroup,
+            string eventHubConnectionString,
+            string avroSchema,
+            int maxBatchSize)
+        {
+            this.executor = executor;
+            this.singleDispatch = singleDispatch;
+            this.logger = logger;
+            this.brokerList = brokerList;
+            this.topic = topic;
+            this.consumerGroup = consumerGroup;
+            this.eventHubConnectionString = eventHubConnectionString;
+            this.avroSchema = avroSchema;
+            this.maxBatchSize = maxBatchSize;
+        }
+
+        public void Cancel()
+        {
+            this.SafeCloseConsumer();
+        }        
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            ConsumerConfig conf;
+            if (string.IsNullOrEmpty(this.eventHubConnectionString))
+            {
+                // Setup raw kafka configuration
+                conf = new ConsumerConfig
+                {
+                    BootstrapServers = this.brokerList,
+                    GroupId = consumerGroup,
+                    StatisticsIntervalMs = 5000,
+                    SessionTimeoutMs = 6000,
+                    AutoOffsetReset = AutoOffsetReset.Earliest,
+                    EnablePartitionEof = true,
```

Comment:
I'm assuming eventually some of these values should be configurable through `host.json`, right?

Reply:
Correct. We will make sure these get moved over in to config files as per the convention with other triggers.

Reply:
We're just trying to setup a starting point here for the team. We can either address this now in this PR, or come back to it later. 

@fbeltrao now, or later?

Reply:
Makes perfect sense, I was just wondering.

Reply:
Thanks for the reminder, I have an [issue ](https://github.com/Microsoft/azure-functions-kafka-extension/issues/16)to make sure we go back and address this.